src/train.py

```diff
@@ -0,0 +1,126 @@
+from absl import flags, app
+from glob import glob
+import numpy as np
+import torch
+from torch import nn
+from torch.utils import data
+from serialization.fontDataset import Dataset
+from sklearn.metrics import confusion_matrix
+import matplotlib.pyplot as plt
+import pantry
+
+np.set_printoptions(threshold=np.inf)  # Print full confusion matrix.
+FLAGS = flags.FLAGS
+flags.DEFINE_string(
+    "pastry", None,
+    "Pastry: indicates network, optimizer and number of epochs")
+flags.mark_flag_as_required("pastry")
+
+DATA_PATH = '/zooper1/fontbakers/data/singleExampleProtos/'
+DIMENSIONS = (20, 30, 3, 2)
+TRAIN_TEST_SPLIT = 0.8
+
+
+def validate(epoch_num):
+    print('\tValidating...')
+
+    valset = Dataset(font_files_val, DIMENSIONS)
+    valloader = data.DataLoader(
+        valset, batch_size=16, shuffle=True, num_workers=4)
+
+    correct = 0
+    total = 0
+    true_labels = []
+    pred_labels = []
+
+    with torch.no_grad():
+        for font, labels in valloader:
+            font = font.float().to(device)
+            labels = [class_index[label] for label in labels]
+            labels = torch.tensor(labels).to(device)
+
+            outputs = net(font)
+            _, predicted = torch.max(outputs.data, dim=1)
+            total += len(labels)
+            correct += (predicted == labels).sum().item()
+            pred_labels.append(predicted.cpu())
+            true_labels.append(labels.cpu())
+
+    pred_labels = torch.cat(pred_labels)
+    true_labels = torch.cat(true_labels)
+
+    confusion = confusion_matrix(true_labels, pred_labels)
+    fig, ax = plt.subplots(figsize=[18, 12])
+    plt.imshow(confusion)
+    plt.colorbar()
+    classes = class_index.keys()
+    tick_marks = np.arange(len(classes))
+    plt.xticks(tick_marks, classes, rotation=45)
+    plt.yticks(tick_marks, classes)
+    plt.ylabel('True label')
+    plt.xlabel('Predicted label')
+    plt.tight_layout()
+    plt.savefig('confusion_{}.png'.format(epoch_num))
```

Comment:
Save to unique dir?

Reply:
What should we name it? I've just been in the habit of appending the latest committish to the names of all output files.